### PR TITLE
Added automatic expansible dialog in order to fit the text of the Titles of the Tab Bar

### LINKED ISSUE #14042: The more info tab bar overflows in some languages on desktop
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When Home Assistant 2022.09 with #13416 was released (which is a great addition btw 🎉!). It seems the additional options within the tab bar can cause the options to be misaligned. In English it looks correct, like this:

English tab bar

However in Dutch it looks like this:

Misaligned tab bar

The options are not centered as I would expect. I have tested all languages and found the following languages with this issue. Although some languages seem to have incomplete translations.

#### Español (Latin America)

Spain

#### Nederlands

Netherlands

#### Português (BR)

Brasil

#### Українська

Ukraine

### Describe the behavior you expected

I expect the tab bar to fit the dialog and to be neatly centered, like in English and many other languages.

### Steps to reproduce the issue

1. Change your language to **Nederlands**
2. Open a more info dialog

### What version of Home Assistant Core has the issue?

2022.10

### What was the last working version of Home Assistant Core?

2022.08

### In which browser are you experiencing the issue with?

Google Chrome 105.0.5195.127

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

I've already looked into this issue myself and I found out that the dialog has a width of `560px` as described by the Material Design 3 spec. At least that is what I could find in #13084 and in [the spec](https://m3.material.io/components/dialogs/specs). That's why I suppose that in #13416 it was set on `560px` 🤷:

https://github.com/home-assistant/frontend/blob/310b110b8b127db88c23cb4873a85719694ae0eb/src/dialogs/more-info/ha-more-info-dialog.ts#L301-L307

However when you would set the ` --mdc-dialog-max-width` to a higher value, such as `600px` most of the languages will fit, but not all. Like this:

image
image

IMHO this is more a workaround 😒. Another solution would be to change the translation to something that is shorter. In Dutch it literally says _Entity Settings_, maybe _Settings_ is more than sufficient, like it is in English:

image
image

It still barely fits and I'm unaware if that would potentially break something else 😐. I already asked the same question in Discord on `devs_frontend`, without a clear answer. That's why I'm opening this issue, as I really would love to contribute and fix this 😄 . But I want to do it the right way and not waste (y)our time.

### PR DESCRIPTION
**Pull request by jrobles98**
## Proposed change

In plain English language the Dialogs looks good and everything fits, but not in other langs. Specifically the titles of the Tab Bar of the Dialog get so big that it overcomes the Dialog element and you will have to scroll horizontally to see the rest of the Tabs (it's a forced feature of the Tab Bar component of Material Design).

So what I did is under those circumstances (having 4 tabs or more) let the Dialog element resize with a dynamic width based on the size available on the screen so that it will **take the minimum needed** to not have to scroll.

It is supposed to not change anything under English language, only in the cases where the translations exceeds the available space.
In order to test this I have tested it in _Nederlands_ and _Español (Latin America)_

This issue was already mentioned here: #14042 

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Simply trying by changing language to either _Nederlands_ or _Español (Latin America)_ and then going to any entity that has more than 3 tabs in the "more info" Dialog.
Then to see that the Dialog is resized to fit all the tittles if there is enough space available
(The maximum allowed to take is `85vw` of the screen)

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #14042 
- This PR is related to issue or discussion: #13084 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Hi, thanks for the PR.

First of all, we should update some translations, `entity` should be removed from all translations (entiteits instellingen => instellingen)

Second, these elements are designed to overflow, they can scroll to the overflow parts

Third, why not make it a little wider everywhere?

**Comment by piitaya** [comment]
And the problem is still here on mobile 😞 (but it's easier to scroll as it's a touch device).

@bramkragten Sometimes, we do not want to have larger dialog because the content is can be on multiple line.

**Comment by bramkragten** [comment]
> @bramkragten Sometimes, we do not want to have larger dialog because the content is can be on multiple line.

But in that case we also don't want it for other languages?

**Comment by jrobles98** [comment]
In order to clarify the visual changes beforehand:

1. This change doesn't increase the standard width of the dialog, it only does when it is needed in other languages and also only taking into account the size of the tab bar.

2. In English there are no changes at all visually, at least as far as I have been able to test this. It was not my intention to change something that already works and looks fine. :)

3. To explain better how this behaves depending on the circumstances: if there are 4 tabs or more it will add the following to the dialog:
    - `min-width` and `max-width` => `560px` and `85vw`.
    - The width of the dialog will always try to fit the minimum space the tab bar can be in.
    - The contents of the dialog's current tab won't change or expand the dialog's width.

If you are still not convinced i can upload some screenshots showing this feature in action

**Comment by alex3305** [comment]
> First of all, we should update some translations, `entity` should be removed from all translations (entiteits instellingen => instellingen)

@bramkragten That's what I also suggested in the linked issue (#14042). But this issue would still not be solved for Brazilian Portuguese unfortunately 😒.

> Second, these elements are designed to overflow, they can scroll to the overflow parts

Yes, on devices with a touch screen that works great, like @piitaya says. But without an horizontal scroll wheel I have a really hard time scrolling on desktop. Also I think it looks a bit off 🤷.

> Third, why not make it a little wider everywhere?

Because that would conflict with the [Material 3 Spec](https://m3.material.io/components/dialogs/specs), which mandates a `560dp` width of dialogs everywhere.

> And the problem is still here on mobile 😞 (but it's easier to scroll as it's a touch device).
This issue should not affect mobile devices (see the media query). Or am I just talking 💩 now?

IMHO the most elegant solution would be to change the translations to _'Settings'_ everywhere, so it's the same as in English. Make the `--mdc-dialog-*-width` themeable with a variable. And set the `--mdc-dialog-max-width` of the dialog to something larger than the current `560px` even if it conflicts with the M3 Spec.

----

@jrobles98 `85vw` can cause even more issues with the media query. If the client width is `601px` and you have a `max-width` of `85vw`, the calculated `max-width` would only be `510px`. Which is less than the current `560px`.

**Comment by jrobles98** [comment]
> @jrobles98 `85vw` can cause even more issues with the media query. If the client width is `601px` and you have a `max-width` of `85vw`, the calculated `max-width` would only be `510px`. Which is less than the current `560px`.

But in this case CSS standards always stays that `min-width` property overrides `max-width`.
So when those screen cases are encountered and there is a conflict between the two, then only `min-width` is going to be taken into consideration.

And BTW I already tested it just to make sure nothing breaks hehe :)

**Comment by bramkragten** [comment]
I think we can safely make the dialog 580px wide, and update the translations. It would be better than adding all this extra logic that has to be maintained.
